tests/testthat/test-dfm_weight.R

```diff
@@ -282,3 +282,19 @@ test_that("docfreq works previously a weighted dfm (#1237)", {
         computed
     )
 })
+
+test_that("smooth slot is correctly set (#1274)", {
```

Comment:
I added these tests, since they should pass for this fix to have worked. They don't pass, which tells me that the `smooth` slot is getting reset somewhere. I would check the `as.dfm()` and the constructor methods for the `dfm` class. 

Test, test, test ... there's a lot to be said for [writing tests with expectations first and then coding](https://en.wikipedia.org/wiki/Test-driven_development) to meet the expectations. 

Also raises a question - what should `@smooth` be if we smooth a dfm by 1 and then smooth that smoothed dfm by 2?

Reply:
:+1: will look into the tests and writing them!

I think if you smooth again, `@smooth` should be `x@smooth + smoothing` so 3 in your example. Because the entries in the dfm of `dfm_smooth(dfm_smooth(x, 1), 2)` should be exactly the same as `dfm_smooth(x, 3)`.

I fixed the code to not reset but add to `@smooth`. (I also fixed the test for this last case to use the dfms1).